rust_src/src/character.rs

```diff
@@ -83,25 +78,23 @@ pub fn char_or_string_p(object: LispObject) -> bool {
 
 /// Convert the byte CH to multibyte character.
 #[lisp_fn]
-pub fn unibyte_char_to_multibyte(ch: LispObject) -> LispObject {
-    let c = ch.as_character_or_error();
-    if c >= 0x100 {
-        error!("Not a unibyte character: {}", c);
+pub fn unibyte_char_to_multibyte(ch: Codepoint) -> Codepoint {
```

Comment:
I don't understand how this stuff works, but would it be possible to further break `Codepoint` into `Unibyte` and `Multibyte` types? Or something like that?

Reply:
It would be possible but I think it would create more problems than it would solve, especially when it comes to byte8 handling